gems/smithy/lib/smithy/model/flattener.rb

```diff
@@ -0,0 +1,68 @@
+# frozen_string_literal: true
+
+module Smithy
+  module Model
```

Comment:
I _think_ we were considering either renaming this module name so it doesn't share namespaces with `smithy-model`  - I tried to dig in [smithy docs](https://smithy.io/2.0/spec/model.html) and internal docs  for ideas:
* Smithy::JsonModel (based on the fact that we are working with JSON AST)
* Smithy::DataModel 
* Smithy::ApiModel (I think you mentioned this previously in a discussion)

Could also go back and rename smithy-model gem into smithy-shapes since the gem contains building blocks to create shapes? Maybe not a good idea since "shapes" are very overloaded....

Reply:
I will rename this module in another commit, probably named API. I think smithy-model is correct for types and shapes.